core/proc/element_reference_spec.rb

```diff
@@ -14,3 +15,12 @@
 describe "Proc#call on a Proc created with Kernel#lambda or Kernel#proc" do
   it_behaves_like :proc_call_on_proc_or_lambda, :call
 end
+
+describe "Proc#[] with frozen_string_literals" do
+  it "doesn't duplicate frozen strings" do
+    proc {|a| a }["sometext"].frozen?.should be_false
```

Comment:
Could you move those to a separate file, with frozen string literals: false, so it doesn't depend on whether this file uses the magic comment?